code/modules/fishing/sources/_fish_source.dm

```diff
@@ -280,7 +283,7 @@ GLOBAL_LIST(fishing_property_cache)
 /datum/fish_source/proc/get_fish_table()
 	var/list/table = fish_table.Copy()
 	for(var/result in table)
-		if(fish_counts[result] == 0)
+		if((result in fish_counts) && fish_counts[result] <= 0)
 			table -= result
```

Comment:
why? also when could these ever desync

Reply:
rewards are rolled before they're spawned. Players usually have to finish a minigame for the reward to spawn, which makes it a possibility. I might devise a way to prevent this by temporarily lowering the value of the reward path in the fish_counts list by one in the future, but I cannot do it right now, since I've another PR up atm that slightly improves this situation already.